src/mapping.ts

```diff
@@ -61,5 +60,44 @@ export function handleTransfer(event: Transfer): void {
   transferEvent.blockNumber = event.block.number.toI32()
   transferEvent.transactionId = event.transaction.hash
   transferEvent.save()
+}
+
+export function handleMetadataChanged(event: MetadataChanged): void {
+  let domain = Domain.load(event.params.id.toHex())
+  if(domain == null) {
+     domain = new Domain(event.params.id.toHex())
+  }
+  domain.metaData = event.params.uri
+  domain.save()
+}
 
+export function handleMetadataLocked(event: MetadataLocked): void {
+  let account = new Account(event.params.locker.toHex())
+  account.save()
+
+  let domain = Domain.load(event.params.id.toHex())
+  if(domain == null) {
+     domain = new Domain(event.params.id.toHex())
+  }
+  domain.isLocked = true
+  domain.lockedBy = account.id
+  domain.save()
+}
+
+export function handleMetadataUnlocked(event: MetadataUnlocked): void {
+  let domain = Domain.load(event.params.id.toHex())
+  if(domain == null) {
+     domain = new Domain(event.params.id.toHex())
+  }
+  domain.isLocked = false
```

Comment:
You're also going to want to clear out the `lockedBy` field and set it to null

Reply:
I may be misunderstanding here but if Im just setting it to null anyways why would I have it marked as !  ?

wouldnt removing the ! in the schema simply make it default to null until its set or am I missing something?

Reply:
This is in reference to if the domain gets unlocked, it means there is no locker, therefore the locker field should be cleared (set to null)